spec/acceptance/test_spec.rb

```diff
@@ -17,5 +17,64 @@
       its(:stdout) { is_expected.to match(%r{Running unit tests}) }
       its(:stderr) { is_expected.not_to match(%r{WARN|ERROR|FAIL}i) }
     end
+
+    context 'listing tests' do
+      before(:each) do
+        spec_file = File.join('spec', 'demo_spec.rb')
+        File.open(spec_file, 'w') do |f|
+          f.puts <<-EOF
+require 'spec_helper'
+
+describe Hash do
+  on_supported_os.each do |os, facts|
+    context "On OS \#{os}" do
+      it 'should return a blank instance' do
+        Hash.new.should == {}
+      end
+    end
+  end
+end
+EOF
+        end
+      end
+
+      describe command('pdk test unit --list') do
+        its(:exit_status) { is_expected.to eq 0 }
+        its(:stdout) { is_expected.to match(%r{Examples:.*demo_spec.rb\[1:1:1\].*demo_spec.rb\[1:2:1\].*demo_spec.rb\[1:3:1\]}m) }
```

Comment:
Rather than using a big multiline regex, I'd suggest doing multiple matches on stdout

```ruby
its(:stdout) { is_expected.to match(%r{^demo_spec\.rb\[1:1:1\]}) }
its(:stdout) { is_expected.to match(%r{^demo_spec\.rb\[1:2:1\]}) }
its(:stdout) { is_expected.to match(%r{^demo_spec\.rb\[1:3:1\]}) }
```